src/common/rabbitmq/rabbitmq.service.ts

```
@@ -0,0 +1,116 @@
+
+      if (!hostname || !port || !username || !password) {
+        Logger.error(
+          "RabbitMQ is enabled but missing one or more config variables.",
+          "RabbitMQService",
+        );
+      } else {
+        this.connectionOptions = {
+          protocol: "amqp",
+          hostname: hostname,
+          port: port,
+          username: username,
+          password: password,
+        };
+
+        amqp.connect(
+          this.connectionOptions,
+          (connectionError: Error, connection: Connection) => {
+            if (connectionError) {
+              Logger.error(
+                "Connection error in RabbitMQService: " +
+                  JSON.stringify(connectionError.message),
+                "RabbitMQService",
+              );
+              return;
+            }
+            this.connection = connection;
+
+            this.connection.createChannel(
+              (channelError: Error, channel: Channel) => {
+                if (channelError) {
+                  Logger.error(
+                    "Channel error in RabbitMQService: " +
+                      JSON.stringify(channelError.message),
+                    "RabbitMQService",
+                  );
+                  return;
+                }
+                this.channel = channel;
+                Logger.log(this.channel);
+              },
+            );
+          },
+        );
+      }
+    } else {
+      Logger.error("RabbitMQ is not enabled.", "RabbitMQService");
+    }
+  }
+
+  connect(queue: string, exchange: string, key: string) {
+    try {
+      this.channel.assertQueue(queue, { durable: true });
+      this.channel.assertExchange(exchange, "topic", {
+        durable: true,
+      });
+      this.channel.bindQueue(queue, exchange, key);
+    } catch (error) {
+      Logger.error(
+        `Could not connect to RabbitMQ queue ${queue} with exchange ${exchange} and key ${key}.`,
+        "RabbitMQService",
+      );
+      Logger.error(error);
```

I would throw the error and log it higher up if needed.